components/brave_rewards/core/common/url_loader.cc

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/core/common/url_loader.h"
+
+#include <array>
+#include <utility>
+
+#include "base/strings/string_util.h"
+#include "brave/components/brave_rewards/core/common/callback_helpers.h"
+#include "brave/components/brave_rewards/core/initialization_manager.h"
+
+namespace brave_rewards::internal {
+
+namespace {
+
+mojom::UrlResponsePtr CreateShutdownResponse(const mojom::UrlRequest& request) {
+  auto response = mojom::UrlResponse::New();
+  response->url = request.url;
+  response->status_code = -1;
+  return response;
+}
+
+}  // namespace
+
+URLLoader::URLLoader(RewardsEngineImpl& engine) : RewardsEngineHelper(engine) {}
+
+URLLoader::~URLLoader() = default;
+
+void URLLoader::Load(mojom::UrlRequestPtr request,
+                     LogLevel log_level,
+                     LoadCallback callback) {
+  CHECK(request);
+
+  if (engine().Get<InitializationManager>().is_shutting_down()) {
+    Log(FROM_HERE) << request->url
+                   << " will not be fetched: shutdown in progress";
+    DeferCallback(FROM_HERE, std::move(callback),
+                  CreateShutdownResponse(*request));
+    return;
+  }
+
+  LogRequest(*request, log_level);
+
+  client().LoadURL(
+      std::move(request),
+      base::BindOnce(&URLLoader::OnResponse, weak_factory_.GetWeakPtr(),
+                     log_level, std::move(callback)));
+}
+
+bool URLLoader::ShouldLogRequestHeader(const std::string& header) {
+  constexpr std::array allowed_headers{"digest", "signature", "accept",
+                                       "content-type"};
+
+  for (auto* name : allowed_headers) {
```

Comment:
You can also specify these in the loop itself:
```cpp
for (auto* name : {"digest", "signature", "accept", "content-type"}) {
  // ...
}
```

Reply:
Or use `std/base::ranges::any_of<>()`.

Reply:
Cool - I think probably the existing variant is the most self-documenting, though.

Reply:
Yeah, that's fine, too. 👍🏻